src/algebra/order/floor.lean

```diff
@@ -678,6 +678,47 @@ sub_nonneg_of_le $ (le_div_iff hb).1 $ floor_le _
 lemma sub_floor_div_mul_lt (a : k) (hb : 0 < b) : a - ⌊a / b⌋ * b < b :=
 sub_lt_iff_lt_add.2 $ by { rw [←one_add_mul, ←div_lt_iff hb, add_comm], exact lt_floor_add_one _ }
 
+lemma fract_div_nat_cast_eq_div_nat_cast_mod {m n : ℕ} :
+  fract ((m : k) / n) = ↑(m % n) / n :=
+begin
+  rcases n.eq_zero_or_pos with rfl | hn, { simp, },
+  have hn' : 0 < (n : k), { norm_cast, assumption, },
+  refine fract_eq_iff.mpr ⟨by positivity, _, m / n, _⟩,
+  { simpa only [div_lt_one hn', nat.cast_lt] using m.mod_lt hn, },
+  { rw [sub_eq_iff_eq_add', ← mul_right_inj' hn'.ne.symm, mul_div_cancel' _ hn'.ne.symm, mul_add,
+      mul_div_cancel' _ hn'.ne.symm],
+    norm_cast,
+    rw [← nat.cast_add, nat.mod_add_div m n], },
+end
+
+lemma fract_div_int_cast_eq_div_int_cast_mod {m : ℤ} {n : ℕ} :
+  fract ((m : k) / n) = ↑(m % n) / n :=
```

Comment:
I think it is true for both arguments integers as
```suggestion
lemma fract_div_int_cast_eq_div_int_cast_mod {m n : ℤ} :
  int.fract ((m : ℚ) / n) = ↑(m.fmod n) / n :=
```

Reply:
I'm very reluctant to do this because:
 * Mathlib contains literally zero API for `int.fmod`,
 * I'm already yak shaving and currently short of time.

As a compromise I have added a comment proposing the `int.fmod` version.

Reply:
Thanks! Certainly doing that is out of scope for this PR, I should have been clearer about that. I thought you'd be interested to know the version which is true though.